### PR TITLE
store and display notebooks in the documentation

### PR DESCRIPTION
**Pull request by fabianp**
This follows the same process to store and update notebooks as JAX:
https://jax.readthedocs.io/en/latest/developer.html#update-documentation

### CODE REVIEW

**Comment by fabianp** [comment]
This is how it looks:

image

**Comment by fabianp** [comment]
one thing to discuss: colab doesn't have (yet) jaxopt installed by default. Should we start the examples with a ```!pip install jaxopt``` block?

**Comment by mblondel** [comment]
> Should we start the examples with a !pip install jaxopt block?

It installs jaxopt locally in the notebook, right? Then I think it's fine indeed.

File: docs/index.rst [hunk 1]
```diff
@@ -56,6 +56,7 @@ Alternatively, it can be be installed from sources with the following command::
    :maxdepth: 2
    :caption: Examples
 
+   notebooks/resnet_flax
```
Comment (mblondel) [inline docs/index.rst:59]: I would create a top-level category called "Notebooks" here.

**Comment by fabianp** [comment]
This is now ready for review. The added pages currently look like this:

Gallery view:
image

Notebook view:
image

Developer docs view:

image

File: docs/notebooks/index.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+
+
+.. raw:: html
+
+    <div class="sphx-glr-thumbcontainer" tooltip="Resnet example with Flax and JAXopt.">
+
+.. only:: html
+
+ .. figure:: /notebooks/thumbnails/resnet_flax.png
```
Comment (mblondel) [inline docs/notebooks/index.rst:20]:
```suggestion
 .. figure:: /notebooks/deep_learning/resnet_flax_thumbnail.png
```
This way, the thumbnails follow the same folder hierarchy.

File: docs/developer.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+===========
+
+Documentation
+-------------
+
+To rebuild the documentation, install several packages::
+
+    pip install -r docs/requirements.txt
+
+And then run from the docs directory::
```
Comment (mblondel) [inline docs/developer.rst:12]:
```suggestion
And then run from the ``docs`` directory::
```

**Review by mblondel — APPROVED** [review]
LGTM apart from thumbnail location. Tell me what you think!

File: docs/notebooks/index.rst (hunk 2 above)
Comment (fabianp) [reply docs/notebooks/index.rst:20]: that clashes with the jupytext --sync command. But we could have them in `/notebooks/deep_learning/thumbnails/file.png`, what do you think?

Comment (mblondel) [reply docs/notebooks/index.rst:20]: Sounds good.

Comment (fabianp) [reply docs/notebooks/index.rst]: done

File: docs/developer.rst (hunk 3 above)
Comment (fabianp) [reply docs/developer.rst]: done

**Pull request merged**